modules/subchannel/doc/content/modules/subchannel/general/subchannel_theory.md

```diff
@@ -297,5 +296,92 @@ In this case, the conservation equations are recast in matrix form and combined
+
+As soon as the big matrix is constructed, the solver will calculate cross-flow resistances to maintain realizability. A distinctive feature of this method is the introduction of a ``weak relaxation'' logic that stabilizes and accelerates convergence of the coupled $mass flow: (\dot{\mathbf{m}})$, $pressure: (\mathbf{P})$, and $crossflow:(\mathbf{w}_{ij})$ fields in a $Q{=}3$ block-nested linear system with matrix blocks $M_{ij}$ and right-hand-side blocks $\mathbf{b}_i$ that represent the individual governing equations.
+
+#### 1. Fast scale estimates
+
+From the axial- and cross-momentum rows, the code forms quick, diagonally preconditioned estimates:
+\begin{equation}
+\begin{aligned}
+\hat{\mathbf m} &= M_{pm}\,\mathbf m, \\
+\hat{\mathbf p} &= \frac{\hat{\mathbf m}}{\operatorname{diag}(M_{pp}) + \varepsilon_p\mathbf 1},\\
+\hat{\mathbf W} &= \frac{M_{wp}\,\hat{\mathbf p} - \mathbf b_w}{\operatorname{diag}(M_{ww}) + \varepsilon_W\mathbf 1},
+\end{aligned}
+\end{equation}
+with small safeguards $\varepsilon_p,\varepsilon_W\sim 10^{-10}$ to avoid division by zero. Using $\hat{\mathbf W}$, the per-channel crossflow sum $\sum_{j} w_{ij}$ is assembled into a vector $\mathrm{sumw_{ij}}_{\mathrm{loc}}$.
+
+#### 2. Robust scale measurements
+
+Two guarded scalars are computed:
+
+\begin{equation}
+\begin{aligned}
+m_{\min} &= \max\big(\min |\mathbf m|,\; 10^{-10}\big),\\
+S_{\max} &= \max\Big(\max |\mathrm{sumw_{ijloc}}|,\; 10^{-10}\Big)
+\end{aligned}
+\end{equation}
+
+Additionally, a mean inter-iteration change for crossflow is formed
+\begin{equation}
+r_{\mathrm{base}} = \operatorname{mean}\big(\big|\mathbf W^{(k)}| - |\mathbf W^{(k-1)}\big|\big),
+\end{equation}
+leading to a relaxation factor
+\begin{equation}
+r = \frac{r_{\mathrm{base}}}{\max(S_{\max}, \varepsilon)} + 0.5,\qquad \varepsilon\sim10^{-10}.
+\end{equation}
+The +0.5 offset biases toward mild under-relaxation.
+
+#### 3. Crossflow resistance inflation
+
+A cross-coupling resistance is estimated and smoothed:
+\begin{equation}
+\begin{aligned}
+\tilde K   &= \frac{S_{\max}}{m_{\min}}, &
+K^\star &= 0.9\,\tilde K + 0.1\,K_{\text{old}}, &
+K       &= r\,K^\star.
+\end{aligned}
+\end{equation}
+After smoothing, the provisional crossflow resistance $K$ is mapped through a piecewise lower-bound function that enforces minimum safe damping levels in specific ranges.
+
+\begin{equation}
+K \rightarrow
+\begin{cases}
+K , & K >= 10, \\
+1.0, & 1 \leq K < 10, \\
+0.5, & 0.01 \leq K < 0.1, \\
+\frac{1}{3}, & 10^{-3} \leq K < 10^{-2}, \\
+K,            & K < 10^{-3}.
```

Comment:
you are missing this case
    if (_added_K < 1e-2 && _added_K >= 1e-3)
      _added_K = 0.1;